nvflare/edge/web/routing_proxy.py

```diff
@@ -25,10 +25,46 @@
 from nvflare.edge.web.models.api_error import ApiError
 from nvflare.edge.web.web_server import FilteredJSONProvider
 
+# This is just a random large prime number used as number of buckets
+PRIME = 100003
+
 log = logging.getLogger(__name__)
 app = Flask(__name__)
 
 
+class UniformHash:
+    """A hash algorithm with uniform distribution. It achieves this with following steps,
+        1. Get a hash value using SHA256
+        2. Map the hash value to a virtual hash table with a large prime number
+        3. Map the virtual bucket to real bucket by using an allocation table
+
+    """
+    def __init__(self, num_buckets: int):
+        self.num_buckets = num_buckets
+        self.num = PRIME // num_buckets
+        self.remainder = PRIME % num_buckets
+
+    def get_num_buckets(self) -> int:
+        return self.num_buckets
+
+    def hash(self, key: str) -> int:
+        # The hash() function changes value every run so SHA256 is used
+        sha_bytes = hashlib.sha256(key.encode()).digest()
+        sha = int.from_bytes(sha_bytes[:8], 'big')
+        virtual_hash = sha % PRIME
+
+        start = 0
```

Comment:
I think this chunk of code can be simplified:
idx = virtual_hash // self.num
if idx < self.num_buckets:
     return idx
else:
    return virtual_hash % self.num_buckets

Reply:
Much simpler. Changed.